### ISSUE TITLE
AMP support?

### ISSUE THREAD

**Issue by samazgor** [lavas-project/jekyll-pwa#13]
How to integrate this for AMP pages?

**Comment by xiaoiver** [comment]
May I ask how do you generate your AMP pages?
Using some Jekyll plugins like [amp-jekyll](https://github.com/juusaw/amp-jekyll)?

**Comment by samazgor** [comment]
I've fixed the issue btw. Have to load `jekyll-pwa-plugin.rb` locally and replace lines from 134 to 144.

I m using [amplify](https://github.com/ageitgey/amplify)

**Comment by xiaoiver** [comment]
What `L134-144` do is simply appending service worker registration code to `<body>`.
So maybe I should not hard code the inject point like this.

**Comment by kalwalt** [comment]
Hi @xiaoiver @samazgor i'm using the plugin with amp but i need also to modify the lines from 134 to 144 because amp not allow to inject javscript code, i would to use the code on my host but not sure if i can build and install the gem on Netlify. i could do the modification in the code if someone help me a bit... :smile: I have not so much experience with Ruby and gem.

**Comment by samazgor** [comment]
@kalwalt here is what i did for my [site](https://sam.azgor.com/).
- First of all include the following code before `</head>` of your theme/ template if it's already not there. See [amp referencec](https://www.ampproject.org/docs/reference/components/amp-install-serviceworker).

`<script async custom-element="amp-install-serviceworker" src="https://cdn.ampproject.org/v0/amp-install-serviceworker-0.1.js"></script>`

- Now install `jekyll-pwa` plugin manually, choose option 1 - [Docs](https://jekyllrb.com/docs/plugins/installation/). 
- Modify the **jekyll-pwa-plugin.rb** file and replace `<script> tag` from [L134](https://github.com/lavas-project/jekyll-pwa/blob/eee28f0d9932dfefad008e0383b706be36c5be4e/lib/jekyll-pwa-plugin.rb#L135) to [L144](https://github.com/lavas-project/jekyll-pwa/blob/eee28f0d9932dfefad008e0383b706be36c5be4e/lib/jekyll-pwa-plugin.rb#L144) with the following amp snippet.

`<amp-install-serviceworker src="https://sam.azgor.com/sw-register.js" layout="nodisplay"></amp-install-serviceworker>`

dont forget to replace `sam.azgor.com` with your actual domain name.

**Comment by kalwalt** [comment]
Thank you @samazgor! Yes basically this is what i tryed but the problem is: how i import the plugin inside jekyll? i created the modified  **jekyll-pwa-plugin.rb** but i didn't understand if i have to add to the plugin in `_config.yml`. i had never  do this before. I am not very smart with Jekyll and Ruby :sob: For this reason i modified completely the plugin deleting the `self.insert_sw_register_into_body` function and his registration. You can see here my [code](https://github.com/kalwalt/jekyll-amp-pwa/blob/master/lib/jekyll-amp-pwa-plugin.rb) in my fork of the gem. 
Basically in this way i inject the code
`<amp-install-serviceworker src="https://path/to/service-worker.js" layout="nodisplay"></amp-install-serviceworker>`
with a [Liquid tag ](https://github.com/kalwalt/kalwalt.github.io/blob/PWA-feature-fix/_layouts/default.html#L19)
For now i only tested in my localhost but i would try soon on my website.

**Comment by kalwalt** [comment]
My modified version of the gem-plugin is on RubyGems [here](https://rubygems.org/gems/jekyll-amp-pwa-plugin)

**Comment by samazgor** [comment]
@kalwalt create a folder on the root and name it `_plugins`. Now copy your ruby file into that folder. [See this](https://jekyllrb.com/docs/plugins/installation/)

**Comment by kalwalt** [comment]
@samazgor i did as you explain, it works as you said... But in the end i prefer with my custom gem. Thank you for the help! Maybe it will be nice to add an option for custom code injection or disable or enable the injection of the script code....
